src/common/database.c

```diff
@@ -2503,42 +2503,59 @@ void dt_database_show_error(const dt_database_t *db)
-          "     the presence of the <i><b>%s</b></i> file in the <i><b>%s</b></i> folder. \n"
-          "     (full pathname: <i><b>%s</b></i>).\n"
-          "\n"
-          "     <u>Caution!</u> Do not delete these files without first checking that there are no more occurrences of darktable, \n"
-          "     otherwise you risk generating serious inconsistencies in your database.\n"
-          "\n"
-          " As soon as you have identified and removed the cause of the lock, darktable will start without any problem.\n"),
-      db->error_other_pid, lck_filename, lck_dirname, lck_pathname);
+          "  3 - If you have done this or are certain that no other instances of darktable are running, \n"
+          "      this probably means that the last instance was ended abnormally. \n"
+          "      Click on the \"delete database lock files\" button to remove the files <i>data.db.lock</i> and <i>library.db.lock</i>.  \n"
+          "\n\n"
+          "      <i><u>Caution!</u> Do not delete these files without first undertaking the above checks, \n"
+          "      otherwise you risk generating serious inconsistencies in your database.</i>\n"),
+      db->error_other_pid);
+
+    gboolean delete_lockfiles = dt_gui_show_standalone_yes_no_dialog(_("error starting darktable"),
+                                        label_text, _("cancel"), _("delete database lock files"));
+
+    if(delete_lockfiles)
+    {
+      gboolean really_delete_lockfiles = dt_gui_show_standalone_yes_no_dialog(_("are you sure?"),
+                                        _("\n  do you really want to delete the lock files?  \n"), _("no"), _("yes"));
+      if(really_delete_lockfiles)
+      {
+        int status = 0;
+
+        char *lck_filename = g_strconcat(lck_dirname, "/data.db.lock", NULL);
+        if(access(lck_filename, F_OK) != -1)
+          status += remove(lck_filename);
+
+        lck_filename = g_strconcat(lck_dirname, "/library.db.lock", NULL);
+        if(access(lck_filename, F_OK) != -1)
+          status += remove(lck_filename);
 
-    dt_gui_show_standalone_yes_no_dialog(_("darktable cannot be started because the database is locked"),
-                                         label_text, _("close darktable"), NULL);
+        if(status==0)
+          dt_gui_show_standalone_yes_no_dialog(_("all fine"), 
+                                        _("\n  Successfully deleted the lock files.  \n  You can now restart darktable  \n"), 
+                                        _("ok"), NULL);
+        else
+          dt_gui_show_standalone_yes_no_dialog(_("oops"), g_markup_printf_escaped(
+                                        _("\n  At least one file could not be removed.  \n"
+                                        "  You may try to delete the files <i>data.db.lock</i> and <i>library.db.lock</i>  \n"
+                                        "  in folder <a href=\"file:///%s\">%s</a> manually.  \n"), lck_dirname, lck_dirname), 
```

Comment:
Given that the path might be quite long, it might be better reworded as "You may try to manually delete the files [...] in folder [...]" (i.e. move the word "manually")

Reply:
good point, I moved the word 'manually' as suggested